docker-compose.yml

```diff
@@ -9,19 +9,28 @@ services:
      - dbdata:/var/lib/postgresql/data
     environment:
     - POSTGRES_PASSWORD=developmentpassword
+  middleman:
+    build:
+      context: .
+      target: middleman
+      cache_from:
+        - ${dockerHubUsername:-dfedigital}/teacher-training-api-middleman:${GIT_BRANCH:-latest}
+    image: ${dockerHubUsername:-dfedigital}/teacher-training-api-middleman:${GIT_BRANCH:-latest}
```

Comment:
Ya, I think I was assuming that the same agent would be running all the steps, so it wouldn't need to have pushed the `master` image (note that we don't pull as part of the "execute tests" step). However, I'm not sure there's a reason _not_ to push the `master`.

I don't think `latest` would ever show up here when running on CI, this would purely be for convenience when running locally so devs can get it working without having to specifying the env var `GIT_BRANCH`. This may be more surprising than convenient, however, time will tell and we can change this at any time.